IdentityServer/v6/docs/content/quickstarts/1_client_credentials.md

```diff
@@ -463,14 +463,23 @@ You can now enforce this policy at various levels, e.g.:
 * for all API endpoints
 * for specific controllers/actions
 
-Typically you setup the policy for all API endpoints in the routing system:
+Typically you setup the policy for all API endpoints in the routing system.
+You'll have to add a call to app.UseRouting() *before* the call the app.UseEndpoints(),
+and *remove* the call to app.MapControllers(). Also, if there are calls to
+*both* app.UseRouting() and app.UseEndpoints(...), the call to app.UseAuthorization() must go between them:
 
 ```cs
+app.UseRouting();
+...
+app.UseAuthorization();
+...
 app.UseEndpoints(endpoints =>
 {
     endpoints.MapControllers()
         .RequireAuthorization("ApiScope");
 });
+
+//app.MapControllers();
```

Comment:
Thank you very much for catching this. I'd like to simplify it further by replace the routing and endpoints with chaining a `RequireAuthorization` call onto the `app.MapControllers()` call. Routing and endpoints works too of course, but chaining onto MapControllers reduces the number of changes to the pipeline compared with what is generated by `dotnet new webapp`, which will be slightly more friendly.